SI-6989 privateWithin is now populated in reflect

xeno-by opened the pull request:
Runtime reflection in JavaMirrors previously forgot to fill in
privateWithin when importing Java reflection artifacts. Now this is fixed.

xeno-by committed:
SI-6989 privateWithin is now populated in reflect

Runtime reflection in JavaMirrors previously forgot to fill in
privateWithin when importing Java reflection artifacts. Now this is fixed.

xeno-by commented:
review @retronym

retronym commented on test/files/run/t6989.scala:
```diff
@@ -0,0 +1,6 @@
+
+object Test extends App {
+  def show(s: Symbol) = s"$s: private=${s.isPrivate} protected=${s.isProtected} privateWithin=${s.privateWithin}"
+  (typeOf[java.io.File].declarations map show filter (_ contains "package ")).toList.sorted foreach println
```
It would be clearer to me to include a `_1.java` source files, rather than piggybacking the test from `java.io.File`. You could then check the variations of Java access.

xeno-by replied:
Done

xeno-by commented:
PLS REBUILD ALL

xeno-by commented:
@retronym Introduced an extensive test.

scala-jenkins commented:
(kitty-note-to-self: ignore 13101219)
:cat: Roger! Rebuilding pr-rangepos-per-commit, pr-checkin-per-commit. :rotating_light:

retronym commented on test/files/run/t6989/JavaClass_1.java:
```diff
@@ -0,0 +1,7 @@
+  private int x = 0;
+  protected String y = "";
+  public float z = 0.0f;
+}
```
Do we need to test modifiers on statics fields, instance/static methods, and on classes?

xeno-by replied:
We probably could, but they all are handled uniformly. If we test everything diligently, the test is going to be huge.

paulp replied:
Breaking news! Computers can assist in these matters!

xeno-by replied:
Done. Uncovered a few bugs in the process.

xeno-by committed:
introduces an exhaustive java-to-scala test

Originally composed to accommodate pull request feedback, this test has
uncovered a handful of bugs in FromJavaClassCompleter, namely:
* SI-7071 non-public ctors get lost
* SI-7072 inner classes are read incorrectly

I'm leaving the incorrect results of FromJavaClassCompleters in the check
file, so that we get notified when something changes there.

xeno-by commented:
@retronym ping

retronym commented:
Can you please reference those new bugs in a comment in the test (to make it clear without reading the history that the behaviour isn't as desired.)?

Then, lgtm.

xeno-by commented:
Done. PLS REBUILD ALL to be sure

scala-jenkins commented:
(kitty-note-to-self: ignore 13150354)
:cat: Roger! Rebuilding pr-rangepos-per-commit, pr-checkin-per-commit for 02ed5fb3, b43ae58f. :rotating_light:

adriaanm commented:
PLS SYNCH

xeno-by commented:
@adriaanm Interesting new command. What does it mean?

adriaanm commented:
it reconciles the status of jenkins jobs for this PR with the commit statuses on github

scala-jenkins commented:
(kitty-note-to-self: ignore 13199429)
:cat: Synchronaising! :pray:

lrytz commented:
how does it know which are the jenkins jobs for a specific PR, where is that stored?

adriaanm commented:
in the jenkins params -- they store the commit sha, mergebranch and PR number

adriaanm commented:
the problem here was that I formatted the commit status description wrong (forgot a space...) so the logic that makes sure all jobs have finished for each commit was broken

this will illustrate..

BUILDLOG?

lrytz commented:
ah so kitty goes through all the builds in jenkins and finds the ones for a given hash?

adriaanm commented:
yep (and the same PR and the same merge-branch)

lrytz commented:
got it, thanks!

adriaanm commented:
the main remaining downside of the kitten is that it's all polling, so it takes 15 min to discover new commands/commits on a PR

would like to make it use push via github notifications and jenkins build triggers, but that's a longer-term project

scala-jenkins commented:
(kitty-note-to-self: ignore 13200997)
pr-rangepos-per-commit:
- 02ed5fb3: [pending](https://scala-webapps.epfl.ch/jenkins/job//251/):  OK Took 111 min. (But waiting for pr-checkin-per-commit)., [pending](https://scala-webapps.epfl.ch/jenkins/job//251/):  OK Took 111 min.(but waiting for pr-checkin-per-commit)., [pending](https://scala-webapps.epfl.ch/jenkins/job//128/):  started., [success](https://scala-webapps.epfl.ch/jenkins/job//251/): Took 111 min., [success](https://scala-webapps.epfl.ch/jenkins/job//128/): Took 126 min., [success](https://scala-webapps.epfl.ch/jenkins/job//128/): : Took 126 min., pending:  queued.
- b43ae58f: [pending](https://scala-webapps.epfl.ch/jenkins/job//275/):  OK Took 80 min. (But waiting for pr-checkin-per-commit)., [pending](https://scala-webapps.epfl.ch/jenkins/job//275/):  OK Took 80 min.(but waiting for pr-checkin-per-commit)., [success](https://scala-webapps.epfl.ch/jenkins/job//275/): Took 80 min., [pending](https://scala-webapps.epfl.ch/jenkins/job//275/):  started., pending:  queued.

pr-checkin-per-commit:
- 02ed5fb3: [success](https://scala-webapps.epfl.ch/jenkins/job//276/):  Took 79 min., [pending](https://scala-webapps.epfl.ch/jenkins/job//276/):  OK Took 79 min.(but waiting for pr-rangepos-per-commit)., [pending](https://scala-webapps.epfl.ch/jenkins/job//276/):  started., [success](https://scala-webapps.epfl.ch/jenkins/job//276/): Took 79 min., pending:  queued.
- b43ae58f: [pending](https://scala-webapps.epfl.ch/jenkins/job//256/):  OK Took 82 min. (But waiting for pr-rangepos-per-commit)., [success](https://scala-webapps.epfl.ch/jenkins/job//256/): Took 82 min., [pending](https://scala-webapps.epfl.ch/jenkins/job//256/):  started., pending:  queued.

xeno-by commented:
PLS SYNCH

scala-jenkins commented:
(kitty-note-to-self: ignore 13226661)
:cat: Synchronaising! :pray:

adriaanm commented:
all these commits have been built -- "pending OK" means it built ok but is waiting for other jobs to finish

there's a bug in the kitten regarding making sure the PR doesn't show up as mergeable until all commits and all jobs finish (since github only looks at the last status of the last commit)

adriaanm commented:
when a PR has the label "tested", it means that its good to go (once it's reviewed)

xeno-by commented:
Looks like we have the "tested" label here, therefore merging.
